src/components/ReadingGoalPage/useReadingGoalReducer.ts

```diff
@@ -0,0 +1,176 @@
+      };
+    }
+  | {
+      type: 'SET_SECONDS';
+      payload: {
+        seconds: ReadingGoalState['seconds'];
+      };
+    }
+  | {
+      type: 'SET_DURATION';
+      payload: {
+        duration: ReadingGoalState['duration'];
+      };
+    }
+  | {
+      type: 'SET_RANGE';
+      payload:
+        | {
+            startVerse: ReadingGoalState['rangeStartVerse'];
+            endVerse: ReadingGoalState['rangeEndVerse'];
+          }
+        | {
+            startVerse: NonNullable<ReadingGoalState['rangeStartVerse']>;
+            endVerse: NonNullable<ReadingGoalState['rangeEndVerse']>;
+          };
+    }
+  | {
+      type: 'SET_EXAMPLE';
+      payload: {
+        exampleKey: NonNullable<ReadingGoalState['exampleKey']>;
+      };
+    };
+
+const reducer = (state: ReadingGoalState, action: ReadingGoalAction): ReadingGoalState => {
+  switch (action.type) {
+    case 'SET_PERIOD':
+      return {
+        ...state,
+        duration: action.payload.period === 'continuous' ? 30 : null,
+        period: action.payload.period,
+      };
+    case 'SET_TYPE':
+      return {
+        ...state,
+        type: action.payload.type,
+      };
+    case 'SET_PAGES':
+      return {
+        ...state,
+        pages: action.payload.pages,
+      };
+    case 'SET_SECONDS':
+      return {
+        ...state,
+        seconds: action.payload.seconds,
+      };
+    case 'SET_DURATION':
+      return {
+        ...state,
+        duration: action.payload.duration,
+        period: action.payload.duration ? 'continuous' : 'daily',
+      };
+    case 'SET_RANGE':
+      return {
+        ...state,
+        rangeStartVerse: action.payload.startVerse,
+        rangeEndVerse: action.payload.endVerse,
+      };
+    case 'SET_EXAMPLE': {
+      const example = readingGoalExamples[action.payload.exampleKey];
+      let values = {};
+      if ('values' in example) {
+        values = { ...example.values };
+      }
+
+      return {
+        ...state,
+        exampleKey: action.payload.exampleKey,
+        ...values,
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+export const readingGoalExamples = {
+  time: {
+    i18nKey: 'time',
+    icon: ClockIcon,
+    recommended: true,
+    values: {
+      type: ReadingGoalType.TIME,
+      seconds: 10 * 60,
+      period: 'daily',
+    },
+  },
+  khatm: {
+    i18nKey: 'khatm',
+    icon: BookIcon,
+    values: {
+      type: ReadingGoalType.RANGE,
+      rangeStartVerse: '1:1',
+      rangeEndVerse: '114:6',
+      duration: 30,
+      period: 'continuous',
+    },
+  },
+  custom: {
+    i18nKey: 'custom',
+    icon: SettingsIcon,
+  },
+} as const;
+
+const initialState: ReadingGoalState = {
+  period: 'daily',
+  type: ReadingGoalType.PAGES,
+  exampleKey: null,
+  pages: 1,
+  seconds: 60,
+  duration: null,
+  rangeStartVerse: '1:1',
+  rangeEndVerse: '114:6',
+};
+
+const useReadingGoalReducer = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
```

Comment:
Is there a reason we are using `useReducer` vs a Redux slice as we have always been doing?

Reply:
just didn't think we needed global state. Do you think we should refactor?

Reply:
It's okay sounds good let's keep `useReducer``